### PR TITLE
Introduce package:sky/animation.dart

### PR DESCRIPTION
**Pull request by abarth**
Move the animation libraries into src/animation and change importers to use
package:sky/animation.dart. Also, move scheduler.dart into the animation
library so that the animation library can be self-contained.

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie @mpcomplete 

I was going to add dartdoc to the animation library but I ended up reorganizing it a bit.  IMHO, it should be one of the top-level libraries we export to developers.

**Comment by Hixie** [comment]
Please update the README accordingly.

File: sky/packages/sky/lib/src/animation/scheduler.dart [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+      return;
+    sky.view.scheduleFrame();
+    _haveScheduledVisualUpdate = true;
+  }
+}
+
+/// Registers [beginFrame] callback with the engine.
+Scheduler _init() {
+  Scheduler scheduler = new Scheduler._();
+  sky.view.setFrameCallback(scheduler.beginFrame);
```
Comment (Hixie) [inline sky/packages/sky/lib/src/animation/scheduler.dart]: why isn't this in the constructor?

File: sky/packages/sky/lib/src/animation/scheduler.dart [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  /// The callback will be run prior to flushing the main rendering pipeline.
+  /// Typically, requestAnimationFrame is used to throttle writes into the
+  /// rendering pipeline until the system is ready to accept a new frame. For
+  /// example, if you wanted to tick through an animation, you should use
+  /// requestAnimation frame to determine when to tick the animation. The callback
+  /// is passed a timeStamp that you can use to determine how far along the
+  /// timeline to advance your animation.
+  ///
+  /// Returns an id that can be used to unschedule this callback.
+  int requestAnimationFrame(SchedulerCallback callback) {
```
Comment (Hixie) [inline sky/packages/sky/lib/src/animation/scheduler.dart]: should we use futures for this? Some subclass of Future that can be canceled, maybe?

File: sky/packages/sky/lib/src/animation/scheduler.dart [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        callback(timeStamp);
+    });
+    _removedIds.clear();
+
+    for (SchedulerCallback callback in _persistentCallbacks)
+      callback(timeStamp);
+  }
+
+  /// Call callback every frame.
+  void addPersistentFrameCallback(SchedulerCallback callback) {
```
Comment (Hixie) [inline sky/packages/sky/lib/src/animation/scheduler.dart]: we should eventually have a way to cancel these, probably

**Comment by Hixie** [comment]
LGTM

File: sky/packages/sky/lib/src/animation/scheduler.dart (hunk 2 above)
Comment (abarth) [inline sky/packages/sky/lib/src/animation/scheduler.dart]: There's only one caller that actually wants to cancel animation frames.  Maybe we should just remove that feature.

**Pull request merged**